core/db/db_test.go

```diff
@@ -31,246 +31,104 @@ func TestMain(m *testing.M) {
-	if err == nil {
-		t.Fatal("Dir alrady exists. DB creation should throw error")
-	}
+func TestOpenDB_NonEmptyDirExists(t *testing.T) {
+	openchainDB := Create()
 	deleteTestDBPath()
+	createNonEmptyTestDBPath()
+
+	defer deleteTestDBPath()
+	defer openchainDB.Close()
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatalf("Failed to open DB: %s", r)
+		}
+	}()
+	openchainDB.Open()
 }
 
-func TestWriteAndRead(t *testing.T) {
-	createTestDB()
-	defer deleteTestDB()
-	performBasicReadWrite(t)
+func TestGetDB_DirDoesNotExist(t *testing.T) {
+	defer deleteTestDBPath()
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatalf("Failed to open DB: %s", r)
+		}
+	}()
+	openchainDB := GetDBHandle()
+
+	defer openchainDB.Close()
 }
 
-func TestOpenDB_DirDoesNotExist(t *testing.T) {
-	deleteTestDBPath()
-	defer deleteTestDB()
-	performBasicReadWrite(t)
+func TestGetDB_NonEmptyDirExists(t *testing.T) {
+	createNonEmptyTestDBPath()
+	defer deleteTestDBPath()
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatalf("Failed to open DB: %s", r)
+		}
+	}()
+	openchainDB := GetDBHandle()
+	defer openchainDB.Close()
 }
 
-func TestOpenDB_DirEmpty(t *testing.T) {
-	deleteTestDBPath()
-	createTestDBPath()
-	defer deleteTestDB()
-	performBasicReadWrite(t)
+func TestWriteAndRead(t *testing.T) {
+	openchainDB := GetDBHandle()
+	defer deleteTestDBPath()
+	defer openchainDB.Close()
+	performBasicReadWrite(openchainDB, t)
 }
 
 // This test verifies that when a new column family is added to the DB
 // users at an older level of the DB will still be able to open it with new code
 func TestDBColumnUpgrade(t *testing.T) {
-	deleteTestDBPath()
-	createTestDBPath()
-	err := CreateDB()
-	if nil != err {
-		t.Fatalf("Error creating DB")
-	}
-	db, err := openDB()
-	if nil != err {
-		t.Fatalf("Error opening DB")
-	}
-	db.CloseDB()
+	openchainDB := GetDBHandle()
+	openchainDB.Close()
 
 	oldcfs := columnfamilies
 	columnfamilies = append([]string{"Testing"}, columnfamilies...)
 	defer func() {
 		columnfamilies = oldcfs
 	}()
-	db, err = openDB()
-	if nil != err {
-		t.Fatalf("Error re-opening DB with upgraded columnFamilies")
-	}
-	db.CloseDB()
-}
-
-func TestDeleteState(t *testing.T) {
-	testDBWrapper := NewTestDBWrapper()
-	testDBWrapper.CreateFreshDB(t)
-	openchainDB := GetDBHandle()
-	defer testDBWrapper.cleanup()
-	openchainDB.Put(openchainDB.StateCF, []byte("key1"), []byte("value1"))
-	openchainDB.Put(openchainDB.StateDeltaCF, []byte("key2"), []byte("value2"))
-	openchainDB.DeleteState()
-	value1, err := openchainDB.GetFromStateCF([]byte("key1"))
-	if err != nil {
-		t.Fatalf("Error getting in value: %s", err)
-	}
-	if value1 != nil {
-		t.Fatalf("A nil value expected. Found [%s]", value1)
-	}
-
-	value2, err := openchainDB.GetFromStateCF([]byte("key2"))
-	if err != nil {
-		t.Fatalf("Error getting in value: %s", err)
-	}
-	if value2 != nil {
-		t.Fatalf("A nil value expected. Found [%s]", value2)
-	}
-}
-
-func TestDBSnapshot(t *testing.T) {
```

Comment:
Why do you want to delete this test? I had added this unit test for increasing the code coverage during unit tests.